### ISSUE TITLE
[Bug report] CatalogKafkaIT > testAlterTopic() FAILED

### ISSUE THREAD

**Issue by danhuawang** [apache/gravitino#4572]
### Version

main branch

### Describe what's wrong

```
> Task :catalogs:catalog-kafka:test

CatalogKafkaIT > testAlterTopic() FAILED
    org.opentest4j.AssertionFailedError: expected: <org.apache.gravitino.client.GenericTopic@60e09cb3> but was: <org.apache.gravitino.client.GenericTopic@1d89e99a>
        at app//org.junit.jupiter.api.AssertionUtils.fail(AssertionUtils.java:55)
        at app//org.junit.jupiter.api.AssertionUtils.failNotEqual(AssertionUtils.java:62)
        at app//org.junit.jupiter.api.AssertEquals.assertEquals(AssertEquals.java:182)
        at app//org.junit.jupiter.api.AssertEquals.assertEquals(AssertEquals.java:177)
        at app//org.junit.jupiter.api.Assertions.assertEquals(Assertions.java:1141)
        at app//org.apache.gravitino.catalog.kafka.integration.test.CatalogKafkaIT.testAlterTopic(CatalogKafkaIT.java:361)

:catalogs:catalog-kafka:test failure marker
  Error: Execution failed for task ':catalogs:catalog-kafka:test':
      org.gradle.api.internal.exceptions.MarkedVerificationException: There were failing tests. See the report at: file:///actions-runner/_work/gravitino-test/gravitino-test/gravitino/build/reports/index.html

> Task :catalogs:catalog-kafka:test FAILED

10 tests completed, 1 failed

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':catalogs:catalog-kafka:test'.
> There were failing tests. See the report at: file:///actions-runner/_work/gravitino-test/gravitino-test/gravitino/build/reports/index.html

* Try:
> Run with --scan to get full insights.

BUILD FAILED in 26m 8s
Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/8.2/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
135 actionable tasks: 78 executed, 57 up-to-date
Error: Process completed with exit code 1.
```

### Error message and/or stacktrace

N/A

### How to reproduce

Github Action:
https://github.com/datastrato/gravitino-test/actions/runs/10443055020/job/28916060692

**Comment by danhuawang** [comment]
@mchades I opened this issue to track the CatalogKafkaIT issue.

**Comment by mchades** [comment]
Thanks for your report! This will be resolved by #4590
